src/me/EtienneDx/RealEstate/Transactions/TransactionsStore.java

```diff
@@ -21,307 +30,685 @@
+                    double signX = rs.getDouble("signX");
+                    double signY = rs.getDouble("signY");
+                    double signZ = rs.getDouble("signZ");
+                    double signPitch = rs.getDouble("signPitch");
+                    double signYaw = rs.getDouble("signYaw");
+                    double price = rs.getDouble("price");
+                    World world = Bukkit.getWorld(signWorld);
+                    if(world == null) continue;
+                    Location sign = new Location(world, signX, signY, signZ, (float) signYaw, (float) signPitch);
+                    IClaim claim = RealEstate.claimAPI.getClaimAt(sign);
+                    if(claim == null) continue;
+                    Player ownerPlayer = (ownerStr != null && !ownerStr.isEmpty()) ? Bukkit.getOfflinePlayer(UUID.fromString(ownerStr)).getPlayer() : null;
+                    // ClaimAuction constructor uses LocalDateTime for endDate.
+                    LocalDateTime endDate = (endDateDate != null) ? LocalDateTime.ofInstant(endDateDate.toInstant(), java.time.ZoneId.systemDefault()) : null;
+                    ClaimAuction ca = new ClaimAuction(claim, ownerPlayer, price, sign, endDate, bidStep);
+                    String buyerStr = rs.getString("buyer");
+                    if(buyerStr != null && !buyerStr.isEmpty()) {
+                    	ca.buyer = UUID.fromString(buyerStr);
+                    }
+                    claimAuction.put(claimId, ca);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+    
+    private void saveDataToDatabase() {
+        String prefix = RealEstate.instance.config.mysqlPrefix;
+        try {
+            // --- Save ClaimSell ---
+            String deleteSell = "DELETE FROM " + prefix + "ClaimSell;";
+            try (PreparedStatement ps = dbConnection.prepareStatement(deleteSell)) {
+                ps.executeUpdate();
+            }
+            String insertSell = "INSERT INTO " + prefix + "ClaimSell (claimId, owner, price, signWorld, signX, signY, signZ, signPitch, signYaw) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?);";
+            try (PreparedStatement ps = dbConnection.prepareStatement(insertSell)) {
+                for (ClaimSell cs : claimSell.values()) {
+                    ps.setString(1, cs.getClaim().getId());
+                    ps.setString(2, cs.getClaim().isAdminClaim() ? "SERVER" : (cs.getOwnerUUID() != null ? cs.getOwnerUUID().toString() : ""));
+                    ps.setDouble(3, cs.getPrice());
+                    ps.setString(4, cs.getSign().getWorld().getName());
+                    ps.setDouble(5, cs.getSign().getX());
+                    ps.setDouble(6, cs.getSign().getY());
+                    ps.setDouble(7, cs.getSign().getZ());
+                    ps.setDouble(8, cs.getSign().getPitch());
+                    ps.setDouble(9, cs.getSign().getYaw());
+                    ps.addBatch();

```

Comment:
I'm surprised java doesn't offer better APIs to do this. IMO this should be moved to its own funciton to keep things as readable as possible